### DISCUSSION TITLE
How to leverage multiprocessing for inference?

### DISCUSSION THREAD

**Discussion by devendrapal5755** [mindee/doctr#833] [Q&A] [answered]
How to use multiprocessing into inference ?

**Comment by fg-mindee** [comment, marked as answer]
Hi @devendrapal5755 :wave: 

Could you elaborate on your question please?
Do you mean performing inference on several pages at the same time for OCR? Do you mean batches of samples for sub-tasks (text detection, text recognition)?

If you meant any of the previous two, our predictor objects will do the job for you!
```python
from doctr.io import DocumentFile
from doctr.models import ocr_predictor

doc = DocumentFile.from_images(["path/to/page1.jpg", "path/to/page2.jpg"])
# The predictor will handle the multiprocessing on python-level, and batching on deep-learning ops
model = ocr_predictor(pretrained=True)
result = model(doc)
```
Let me know if I misunderstood your question!

**Reply by devendrapal5755** [reply]
performing inference on several pages at the same time for OCR its use multiprocessing technique or not , if yes so please explain me.

**Reply by fg-mindee** [reply]
? well, as I just said, it is supported and done by default with the script above :sweat_smile: 

Under-the-hood:
- each predictor chunk the input in batches
- it preprocesses those using multi-threading
- the model performs inference on the batch

If that doesn't answer your question, would you mind being more specific about what you're looking for please? :pray:

**Reply by devendrapal5755** [reply]
Thank you so much, I got it.
